src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -944,3 +1072,51 @@ master_stage_shard_placement_row(PG_FUNCTION_ARGS)
 
 	PG_RETURN_VOID();
 }
+
+
+/*
+ * TableReferenced function checks whether given table is referenced by another table
+ * via foreign constraints. If it is referenced, this function returns true. To check
+ * that, this function searches given relation at pg_constraints system catalog. However
+ * since there is no index for the column we searched, this function performs sequential
+ * search, therefore call this function with caution.
+ */
+bool
+TableReferenced(Oid relationId)
```

Comment:
Soon, we will need this function to use at other places. For example we should add a check to mark_tables_colocated which will use this function.